### ISSUE TITLE
SMS Doesn't Send if the device is dual sim and the sim is on SIM 2 slot

### ISSUE THREAD

**Issue by gitfsms** [SilenceIM/Silence#467]
Message is pending forever if you send on a dual sim device with the sim is only on SIM  2 Slot and nothing on SIM 1..

Tried on Android Jelly bean and Android KitKat dual sim devices

**Comment by johanw666** [comment]
Dual sim is only supported for Android 5 and above. Before that, dual sim solutions were vendor specific and it is nearly impossible to support them all.

**Comment by gitfsms** [comment]
Nope. Try Android 5.2 device and put the sim on SIM 2 and leave the SIM 1 empty.. It won't send and will remain sending forever..

**Comment by johanw666** [comment]
I can't test it anyway, my only dual sim device has Android 4.2 and unfortunately it died a year ago.

**Comment by gitfsms** [comment]
What is happening in 5.2 is also happening down to 4.4 and 4.1. You can't use it to any dual sim phone if the  SIM slot is empty and you are using SIM 2.

Actually I  did a quick look inside made some modification.. The problem as I understand was it won't proceed because it keeps on waiting for a signal/reception from SIM 1(I think) only and re-sent when it is capable to send  without checking another sim slot..

I disabled ServiceRequirement.java by making it return always true and let the SmsSendJob.java determine  if the message was sent or not as usual and it works fine..

I don't know what are the other uses of ServiceRequirement.java in some part of the code I believe it will make the auto-resend feature useless as SmsSendJob.java  will make it failed instantly.

**Comment by noraj1337** [comment]
Just to add that doesn't work either with a Wiko Cink Slim under Android 4.4

**Comment by johanw666** [comment]
@noraj1337 : AFAIK dual SIM suppoirt is only supported for Android >= 5.
